src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/bluetooth/BluetoothManager.kt

```diff
@@ -27,80 +27,157 @@ import kotlinx.coroutines.withTimeoutOrNull
 
 @ExperimentalCoroutinesApi
 class BluetoothManager {
-    private val bluetoothAdapter = BluetoothAdapter.getDefaultAdapter()
+  private val bluetoothAdapter = BluetoothAdapter.getDefaultAdapter()
 
-    suspend fun getBluetoothDevice(discriminator: Int): BluetoothDevice? {
-        val scanner = bluetoothAdapter.bluetoothLeScanner ?: run {
-            Log.e(TAG, "No bluetooth scanner found")
-            return null
-        }
+  suspend fun getBluetoothDevice(discriminator: Int): BluetoothDevice? {
+    val scanner = bluetoothAdapter.bluetoothLeScanner ?: run {
+      Log.e(TAG, "No bluetooth scanner found")
+      return null
+    }
+
+    return withTimeoutOrNull(10000) {
+      callbackFlow {
+        val scanCallback = object : ScanCallback() {
+          override fun onScanResult(callbackType: Int, result: ScanResult) {
+            val device = result.device
+            Log.i(TAG, "Bluetooth Device Scanned Addr: ${device.address}, Name ${device.name}")
 
-        return withTimeoutOrNull(10000) {
-            callbackFlow {
-                val scanCallback = object : ScanCallback() {
-                    override fun onScanResult(callbackType: Int, result: ScanResult) {
-                        val device = result.device
-                        Log.i(TAG, "Bluetooth Device Scanned Addr: ${device.address}, Name ${device.name}")
-
-                        offer(device)
-                    }
-
-                    override fun onScanFailed(errorCode: Int) {
-                        Log.e(TAG, "Scan failed $errorCode")
-                    }
-                }
-
-                val scanFilter = ScanFilter.Builder()
-                    .setServiceData(
-                      ParcelUuid(UUID.fromString(CHIP_UUID)),
-                      byteArrayOf(0, discriminator.toByte(), (discriminator shr 8).toByte())
```

Comment:
This is the correct version except it's missing the vendor and product. We need to keep this ordering and fix ESP32 instead.

https://github.com/project-chip/connectedhomeip/pull/3557

Reply:
Thanks, will revert this change then.

Reply:
Done